lib/ex_aws/request/url.ex

```diff
@@ -106,6 +105,9 @@ defmodule ExAws.Request.Url do
 
   def uri_encode(url), do: URI.encode(url, &valid_path_char?/1)
 
+  defp query_replace_spaces(nil), do: nil
+  defp query_replace_spaces(query), do: String.replace(query, "+", "%20")
```

Comment:
It's `%2B` in one place and `%20` in another. This seems wrong?

Reply:
Actually, that whole `String.replace(new_path, "+", "%2B")` isn't needed and has no effect. The `new_path` gets uri_encoded a little higher up.

In the S3 operation code path, the `url` that gets passed to sanize has already had the query params encoded, but the reset of the URL hasn't been. So if `new_path` in this case contains a +, then it is actually a + and not a space.

s3.ex:29 calls `ExAws.Request.Url.build(config)` which is what encodes the query and not the path. 

It feels a little bit inconsistent to encode one earlier and the other later. Maybe it would be better to shift the encoding to one place in the code. And perhaps it's best to encode both the path and the query in Url.sanitize() because ExAws.Operation.{JSON,Query,RestQuery} also call ExAws.Request.Url.build and therefore have their query encoded if they include any.

Reply:
I've pushed a quick change taking out that replace. Let me know if you think it'd be a good idea for me to try to move the uri_encoding out of `ExAws.Request.Url.build(config)` and to encode both the path and query in ExAws.Request.Url.sanitize instead.

Reply:
`query_replace_spaces` appears to actually be replacing +'s rather than spaces .. and I think that could more simply use `URI.encode_www_form/1` instead of `URI.encode`.

On a related note, the query string parse is redundant with the `URI.parse/1` call which also separates out the query string.

It would be nice to see this fix get into the repo, but perhaps with a bit more cleanup?